api/envoy/extensions/filters/http/credential_injector/v3/credential_injector.proto

```diff
@@ -0,0 +1,113 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.credential_injector.v3;
+
+import "envoy/config/core/v3/http_uri.proto";
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.credential_injector.v3";
+option java_outer_classname = "CredentialInjectorProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/credential_injector/v3;credential_injectorv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Credential Injector]
+// [#not-implemented-hide:]
+// Credential Injector :ref:`configuration overview <config_http_filters_credential_injector>`.
+// [#extension: envoy.filters.http.credential_injector]
+
+// Filter config.
+message CredentialInjector {
```

Comment:
Hi @zhaohuabing , should we support `matcher` as well so that filter can inject credentials only the matched requests?

Reply:
Would prefer just supporting per-route configuration instead of matcher in a filter config.